flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSinkITCase.java

```diff
@@ -426,6 +424,49 @@ public void testNullEnforcer() throws ExecutionException, InterruptedException {
         }
     }
 
+    @Test
+    public void testFromValuesWatermarkPropagation() throws Exception {
+        final StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env);
+        final SharedReference<List<Long>> watermarks = sharedObjects.add(new ArrayList<>());
+        final TableDescriptor sinkDescriptor =
+                TableFactoryHarness.newBuilder()
+                        .schema(Schema.newBuilder().column("a", "INT").build())
```

Comment:
you can use an inline descriptor via `.executeInsert(TableDescriptor)` and safe defining the schema twice

Reply:
Sorry Timo, I don't get what you mean, I'm already using `.executeInsert(sinkDescriptor)`.
Shouldn't the schema be defined both for source and sink?

Reply:
sorry, I wasn't clear enough: you are not using this descriptor as a source, so you can simply omit the schema, it will be derived automatically by `executeInsert`.

Reply:
sure, thx!!!